provisioner/src/lib.rs

```diff
@@ -573,8 +633,23 @@ impl Provisioner for MyProvisioner {
                     .await?
             }
             DbType::AwsRds(AwsRds { engine }) => {
-                self.delete_aws_rds(&request.project_name, &engine.expect("engine to be set"))
-                    .await?
+                let instance_name = format!(
+                    "{}-{}",
+                    &request.project_name,
+                    engine.expect("engine should be set")
+                );
+
+                // Try to delete the subscription item first, if a subscription exists.
+                self.delete_subscription_item(
+                    claim
+                        .token()
+                        .expect("the token should be set on the claim in the jwt auth layer"),
+                    &instance_name,
+                )
+                .await
+                .map_err(|err| Status::internal(err.to_string()))?;
+
+                self.delete_aws_rds(&instance_name).await?
```

Comment:
It seems that if we fail to delete the subscription (which could be something like "we sent a request, it was executed, but we got an error along the way because of the network"), we're going to have deleted a subscription but the RDS instance will be left.

I'm not sure how problematic this is (it's late), but this might need more scrutiny.

Reply:
If there is an error deleting the subscription, e.g. network error or failure to connect to Stripe, we will not try to delete the RDS instance. There are some cases where we will return 200, but a subscription item was not deleted, I explained that a bit more in another comment.

Reply:
My point was the other way around this time. We've successfully deleted the subscription, but a network error made the `delete_subscription_item` call fail here. This is an edge case though, we can disregard that.

Reply:
Ah, you mean the subscription deletion succeeds, but the database deletion fails? Yes, we do not cover that case. We could try to re-add the subscription item in that case, which should be fine since it's prorated to the second, it shouldn't incur any extra cost.

Reply:
That's one way to do it, but I think it's still prone to the network or the app shutting down mid function. Ideally we should make the operation idempotent, log (i.e. save it in a database) that we want to do the operation first, update it when done. That way in the case of an event, we can more easily recover since we know something was started. But again, this is just later, we're not there yet :p

Reply:
Ah, I see what you mean. Yes, that should be implemented eventually, I'll make a note of it! 

I implemented the attempt to recreate the subscription item if it was deleted and the db deletion failed in the latest commit.

Reply:
As discussed on Discord, I dropped the change to try to add back the item.